NotebookAnalyzers/NotebooksAnalysisResults.ipynb

```diff
@@ -0,0 +1,247 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data=[]\n",
+    "for filename in os.listdir(\"../TargetNotebooks/\"):\n",
+    "       if filename.endswith(\".ipynb\"):\n",
+    "              py_code = utils.notebook_to_code(filename)\n",
+    "              notebook_dict = utils.notebook_to_json(filename)\n",
+    "              distribution = utils.markdown_distribution(notebook_dict)\n",
+    "              if utils.markdown_distribution(notebook_dict) is not None:\n",
+    "                     notebook_tuple=(filename,\n",
+    "                            utils.cells_correct_order(notebook_dict),\n",
+    "                            utils.classes_number(py_code),\n",
+    "                            utils.functions_number(py_code),\n",
+    "                            utils.imports_correct_position(py_code),\n",
+    "                            utils.markdown_lines(notebook_dict),\n",
+    "                            utils.markdown_titles(notebook_dict),\n",
+    "                            distribution[0],\n",
+    "                            distribution[1],\n",
+    "                            distribution[2],\n",
+    "                            distribution[3],\n",
+    "                            utils.not_executed_cells(notebook_dict),\n",
+    "                            utils.empty_cells(notebook_dict)\n",
+    "                            )\n",
+    "              else:\n",
+    "                     notebook_tuple=(filename,\n",
+    "                            utils.cells_correct_order(notebook_dict),\n",
+    "                            utils.classes_number(py_code),\n",
+    "                            utils.functions_number(py_code),\n",
+    "                            utils.imports_correct_position(py_code),\n",
+    "                            utils.markdown_lines(notebook_dict),\n",
+    "                            utils.markdown_titles(notebook_dict),\n",
+    "                            None,\n",
+    "                            None,\n",
+    "                            None,\n",
+    "                            None,\n",
+    "                            utils.not_executed_cells(notebook_dict),\n",
+    "                            utils.empty_cells(notebook_dict)\n",
+    "                            )\n",
+    "              data.append(notebook_tuple)\n",
+    "              os.remove(filename[:-5]+\"py\")\n",
+    "              continue\n",
+    "       else:\n",
+    "              continue"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Tutte le osservazioni riguardanti un notebook sono contenute in un'unica tupla, il dataframe viene dunque creato come una lista di tuple, assegnando i corrispondenti nomi alle osservazioni sulle colonne. Ogni riga è un target notebook."
```

Comment:
Anche qui, questo potrebbe benissimo essere testo semplice. I titoli MD solitamente sono più brevi e si usano per delimitare sezioni distinte all'interno di un notebook.

Ad esempio, in un notebook in cui faccio il training di un classificatore, potrei avere le sezioni "Import data", "Prepare data", "Feature extraction", "Model training" e "Model evaluation". Poi, all'interno delle singole sezioni, descrivo discorsivamente con testo MD semplice ciò che faccio, come hai fatto tu in questo caso.

Reply:
Ho cercato di modificare il notebook in modo da rendere i titoli più concisi ed esplicativi e reso normale markdown quelli che prima erano i titoli